### ISSUE TITLE
normalize seems to override coerce callback result

### ISSUE THREAD

**Issue by rpl** [yargs/yargs-parser#56]
It seems that when the two property are combined, `normalize` overrides the coerced value with the normalized version of  the original value of the option.

Minimal nodejs snippet to reproduce the issue:

```
var yargs = require('yargs');
var path = require('path');

var argv = yargs.option('source-dir', {
  normalize: true,
  coerce: path.resolve
}).argv;

console.log("result", {
  sourceDir: argv.sourceDir,
  resolvedSourceDir: path.resolve(argv.sourceDir)
});
```

```
$ node test-normalize-coerce.js --source-dir ../src/../src
result { sourceDir: '../src',
  resolvedSourceDir: '/resolved/dir/src' }
```

**Comment by maxrimue** [comment]
Hey, thanks for reporting! I suppose for now you can instead do something like this:
```javascript
var argv = yargs.option('source-dir', {
  coerce: val => path.resolve(path.normalize(val))
}).argv;
```

**Comment by maxrimue** [comment]
I think #57 squashed the bug found here

**Comment by bcoe** [comment]
@maxrimue @rpl yeah, #57 should fix this; let's work hard to get an update out to yargs this weekend; I think we might need to make it a major, since our coercion is significantly different.

**Comment by maxrimue** [comment]
@bcoe I'd say we add a part about the handling of objects in coercion to the readme at yargs when Greenkeeper sends a pull request for `yargs-parser` once we published the change in the next major :+1:

**Comment by bcoe** [comment]
@maxrimue @rpl #57 is released in `yargs-parser@4.0.0`; @maxrimue feel free to add that doc update 👍
